lib/crossing.rb

```diff
@@ -9,26 +9,27 @@ def initialize(s3_client)
   end
 
   def put(bucket, filename)
-    begin
-      file = File.new(filename, 'r')
-    rescue
-      raise CrossingFileNotFoundException, "File not found: #{filename}"
+    File.open(filename, 'r') do |file|
+      put_content(bucket, File.basename(filename), file.read)
```

Comment:
File.basename returns the filename, but handles Windows path separators transparently